src/test/java/QueryInformationTest.java

```diff
@@ -0,0 +1,146 @@
+    Assert.assertEquals(GMT_TIMEZONE, queryInformation.getTimeZone());
+  }
+
+  @Test
+  public void validCategories() throws ParseException {
+    QueryInformation queryInformation =
+        new QueryInformation(CST_TIMEZONE_ID, CATEGORIES, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(EXPECTED_CATEGORIES, queryInformation.getCategories());
+  }
+
+  @Test
+  public void categoriesLowercasedAndWhitespaceRemoved() throws ParseException {
+    QueryInformation queryInformation = new QueryInformation(
+        CST_TIMEZONE_ID, EDGE_CASE_CATEGORIES, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(EXPECTED_CATEGORIES, queryInformation.getCategories());
+  }
+
+  @Test
+  public void invalidCategories() throws ParseException {
+    // Test with empty string.
+    QueryInformation queryInformation = new QueryInformation(
+        CST_TIMEZONE_ID, EMPTY_STRING, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(EMPTY_IMMUTABLE_SET, queryInformation.getCategories());
+
+    // Test with null.
+    queryInformation =
+        new QueryInformation(CST_TIMEZONE_ID, NULL_VALUE, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(EMPTY_IMMUTABLE_SET, queryInformation.getCategories());
+  }
+
+  @Test
+  public void valideDateRangeParse() throws ParseException {
+    QueryInformation queryInformation =
+        new QueryInformation(CST_TIMEZONE_ID, CATEGORIES, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(JUNE_1_2020_START_OF_DAY, queryInformation.getStartTimestamp());
+    Assert.assertEquals(JUNE_30_2020_END_OF_DAY, queryInformation.getEndTimestamp());
+  }
+
+  @Test(expected = ParseException.class)
+  public void invalidDateRangeEmptyString() throws ParseException {
+    QueryInformation queryInformation = new QueryInformation(
+        CST_TIMEZONE_ID, CATEGORIES, EMPTY_STRING, STORE, MIN_PRICE, MAX_PRICE);
+  }
+
+  @Test
+  public void storeCorrectlySet() throws ParseException {
+    QueryInformation queryInformation =
+        new QueryInformation(CST_TIMEZONE_ID, CATEGORIES, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(EXPECTED_STORE, queryInformation.getStore());
+  }
+
+  @Test
+  public void storeLowercasedAndWhitespaceRemoved() throws ParseException {
+    QueryInformation queryInformation = new QueryInformation(
+        CST_TIMEZONE_ID, CATEGORIES, DATE_RANGE, EDGE_CASE_STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(EXPECTED_STORE, queryInformation.getStore());
+  }
+
+  @Test
+  public void pricesCorrectlySet() throws ParseException {
+    QueryInformation queryInformation =
+        new QueryInformation(CST_TIMEZONE_ID, CATEGORIES, DATE_RANGE, STORE, MIN_PRICE, MAX_PRICE);
+    Assert.assertEquals(EXPECTED_MIN_PRICE, queryInformation.getMinPrice(), ERROR_THRESHOLD);
+    Assert.assertEquals(EXPECTED_MAX_PRICE, queryInformation.getMaxPrice(), ERROR_THRESHOLD);
+  }
+
+  @Test(expected = NumberFormatException.class)
```

Comment:
I asked Ava to do this too, can we use assertThrows(). We want to break your test up into three parts: arrange, act, and assert. You may need to upgrade your version of junit for this.

https://stackoverflow.com/questions/40268446/junit-5-how-to-assert-an-exception-is-thrown
https://howtodoinjava.com/junit5/expected-exception-example/
https://github.com/googleinterns/step12-2020/pull/31/commits/de5cd78b6d147bb98a7e05a9a5da49e7a586678d